src/middleware/middleware-login.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response, NextFunction } from 'express';
+import HttpStateCodes from '../utils/http-state-codes';
+
+export const LoginMiddleware = (req: Request, res: Response, next: NextFunction) => {
+  const { email, password } = req.body
+  
+  const expectedFields = ['email','password'];
+  const validKeys = Object.keys(req.body);
+
+  const missingFields = expectedFields.filter((field) => !validKeys.includes(field));
+
+  const extraFields = validKeys.filter((key) => !expectedFields.includes(key));
+
+  if (missingFields.length > 0) {
+    return res.json({
+      message:`Faltan los siguientes campos en la solicitud: ${missingFields.join(', ')}`, 
+      status:HttpStateCodes.BAD_REQUEST});
+  }
```

Comment:
Change the format of the message in this case the http status code must be write of this form.
```
response.status(HttpStateCodes.BAD_REQUEST).json({message:`Faltan los siguientes campos en la solicitud: ${missingFields.join(', ')}`});
```
Another tip.
Organizing response messages in a separate file is a great way to manage and maintain your code, especially in a TypeScript project. Here's an approach you can use:

1. In a separated file you should create /src/busines/messages.ts 
2. Organize your messages in a way that makes sense for your application. You could categorize them based on features, error types, etc
3. Import the messages into the files where you need them and use them as required.
```
// messages.ts
// this is only an example
const LoginMessages = {
    tokenMissing: `The following fields are missing in the request: ${missingFields.join(', ')}`'
};

const userMessages = {
    userNotFound: 'User not found',
    userCreated: 'User successfully created',
    // ... other user management related messages
};

export { authMessages, userMessages };
```
Please use only english in your messages
How you can use.

```
response.status(HttpStateCodes.UNAUTHORIZED).json({authMessages.tokenMissing});
```